### ISSUE TITLE
Stream upload

### ISSUE THREAD

**Issue by shuhei** [zalando-incubator/perron#102]
## Problem

Currently perron writes a request body to the wire by `requestObject.write(body);`, which doesn't allow us to pipe a readable stream into it. When we want to send a readable stream as a request body (for example, proxying media upload), we have to buffer the content of the readable stream into a `Buffer` and pass it to perron. However, this is not efficient for latency and memory consumption.

## Proposal

Allow users to pass a readable stream as a value of the `body` option. Then perron can pipe the readable stream into the request object.

```ts
if (body && typeof body.pipe === "function") {
  // TODO: error handling...
  body.pipe(requestObject);
} else if (body) {
  requestObject.write(body);
}
```

Error handling should be done at least for:

- error on the `body` readable stream: abort the request
- error on the request: stop reading the `body` readable stream and close it (`readable.destroy()`?)

**Comment by shuhei** [comment]
CC: @prawn-cake

**Comment by prawn-cake** [comment]
As an example:
As a service develop I'd like to use perron to stream `multipart/form` data such as mix of binary and json data (e.g using [form-data](https://www.npmjs.com/package/form-data) which relies on node streams) to the HTTP media server

**Comment by grassator** [comment]
I think that perron should support streams for both request and response body. I tried to look at response body and given current internals error handling is a bit of a nightmare and needs good spring cleaning. Request body should be a bit simpler though so a good place to start.

**Comment by shuhei** [comment]
Published as v0.11.4
